src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +10,43 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
         {
             BoundExpression rewrittenReceiver = VisitExpression(node.ReceiverOpt);
 
-            // field access of a tuple actually accesses the underlying field.
-            FieldSymbol field = (node.FieldSymbol as TupleFieldSymbol)?.UnderlyingTypleFieldSymbol ??
-                                node.FieldSymbol;
+            var expandedNode = ExpandTupleAccessor(node, rewrittenReceiver);
 
-            return MakeFieldAccess(node.Syntax, rewrittenReceiver, field, node.ConstantValue, node.ResultKind, node.Type, node);
+            return MakeFieldAccess(expandedNode.Syntax, expandedNode.ReceiverOpt, expandedNode.FieldSymbol, expandedNode.ConstantValue, expandedNode.ResultKind, expandedNode.Type, expandedNode);
+        }
+
+        /// <summary>
+        /// Converts access to a tuple instance into access into the underlying ValueTuple(s).
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundFieldAccess ExpandTupleAccessor(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = node.FieldSymbol as TupleFieldSymbol;
+            if (tupleField == null)
+            {
+                // replace receiver
+                return node.Update(rewrittenReceiver, node.FieldSymbol, node.ConstantValue, node.ResultKind, node.Type);
+            }
```

Comment:
It doesn't look like there is a good reason to allocate a node that will be right away dropped on the floor. I think we shouldn't even call this method for non-tuple case.